### PR TITLE
feature: add sync state for devops

### PR DESCRIPTION
**Pull request by shaowenchen**
Signed-off-by: shaowenchen <mail@chenshaowen.com>

**What type of PR is this?**
 /kind feature

**What this PR does / why we need it**:

add sync status to annotation for pipeline

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -30,4 +30,4 @@ coverage.txt
 kustomize/network/etcd
 apiserver.local.config
 tmp/
-
+kubesphere.yaml
```
Comment (LinuxSuRen) [inline .gitignore:33]: Where this file comes from? Do any programs generate it?

File: pkg/models/devops/devops.go [hunk 2]
```diff
@@ -173,6 +173,8 @@ func (d devopsOperator) CreateDevOpsProject(workspace string, project *v1alpha3.
 	}
 	project.Name = ""
 	project.Labels[tenantv1alpha1.WorkspaceLabel] = workspace
+	project.Annotations[devopsv1alpha3.DevOpeProjectSyncStatusAnnoKey] = StatusPending
+	project.Annotations[devopsv1alpha3.DevOpeProjectSyncTimeAnnoKey] = GetSyncNowTime()
```
Comment (LinuxSuRen) [inline pkg/models/devops/devops.go:177]: Please consider putting line 176 and line 177 into a function instead of writing them four times.

**Review by LinuxSuRen — COMMENTED** [review]
I guest this feature cannot work alone without the support of the front-end. In order to make it easy to track down. I'd suggest you create an issue in [ks console](https://github.com/kubesphere/console/issues).

I left some comments which are not blocked.

File: pkg/models/devops/devops.go (hunk 2 above)
Comment (shaowenchen) [reply pkg/models/devops/devops.go:177]: It is not a good idea to optimize in advance. There's only one line of code.

File: .gitignore (hunk 1 above)
Comment (shaowenchen) [reply .gitignore:33]: It is a configure file for the developer.  Can we provide a template file? @yuswift

Comment (yuswift) [reply .gitignore:33]: Which folder do you suggest we put it in?

Comment (shaowenchen) [reply .gitignore:33]: add `kubesphere.yaml.tpl` file to this repo, so the developer can just rename it and configure vars to run the project.

```yaml
cat kubesphere.yaml.tpl 
kubernetes:
  kubeconfig: "/root/.kube/config"
  master: https://192.168.0.5:6443
  $qps: 1e+06
  burst: 1000000
authentication:
  authenticateRateLimiterMaxTries: 10
  authenticateRateLimiterDuration: 10m0s
  loginHistoryRetentionPeriod: 168h
  maximumClockSkew: 10s
  multipleLogin: False
  kubectlImage: kubesphere/kubectl:v1.0.0
  jwtSecret: "xxx"
...
```

https://github.com/kubesphere/community/blob/master/developer-guide/development/quickstart.md#understand-kubesphere-cores-configuration

Comment (yuswift) [reply .gitignore:33]: Good suggestion.

**Pull request merged**